import_flow_events.py

```diff
@@ -84,18 +130,81 @@
     CREDENTIALS 'aws_access_key_id={aws_access_key_id};aws_secret_access_key={aws_secret_access_key}'
     FORMAT AS CSV;
 """
-
-Q_CHECK_FOR_DAY = """
-    SELECT timestamp FROM flow_events
-    WHERE timestamp >= {timestamp}
-    AND timestamp < {timestamp} + 86400
-    LIMIT 1;
+Q_INSERT_METADATA = """
+    WITH durations AS (
+      SELECT flowId, MAX(flowTime)
+      FROM flow_csv
+      GROUP BY flowId
+    )
+    INSERT INTO flow_metadata (
+      flowId,
+      beginTime,
+      duration,
+      completed,
+      newAccount,
+      browser,
+      browserVersion,
+      os,
+      context,
+      entrypoint,
+      migration,
+      service,
+      utmCampaign,
+      utmContents,
+      utmMedium,
+      utmSource,
+      utmTerm
+    )
+    SELECT (
+      begin.flowId,
+      (begin.timestamp * 1000.0),
+      (durations.flowTime * 1000.0),
+      (CASE WHEN signed.flowId IS NULL THEN FALSE ELSE TRUE END),
+      (CASE WHEN created.flowId IS NULL THEN FALSE ELSE TRUE END),
+      begin.browser,
+      begin.browserVersion,
+      begin.os,
+      begin.context,
+      begin.entrypoint,
+      begin.migration,
+      begin.service,
+      begin.utmCampaign,
+      begin.utmContents,
+      begin.utmMedium,
+      begin.utmSource,
+      begin.utmTerm
+    )
+    FROM flow_csv AS begin
+    INNER JOIN durations
+      ON begin.flowId = durations.flowId AND begin.type = 'flow.begin'
+    LEFT JOIN flow_csv AS created
+      ON begin.flowId = created.flowId AND created.type = 'account.created'
+    LEFT JOIN flow_csv AS signed
+      ON begin.flowId = signed.flowId AND signed.type = 'account.signed';
```

Comment:
If the joins here get too unwieldy, another option could be to update the table piecemeal, along the lines of:

1. Insert a new row for each `flow.begin` event in the import table, defaulting to existing-account, incomplete, etc.
2. Do an update to set the duration for each flowid in the import table.
3. Do an update to set `newAccount=1` for all `account.created` events in the import table.
4. Do an update to set `completed=1` for all `account.signed` events in the import table.

This might make it easier to handle flows that cross the boundary between days, e.g. if the `flow.begin` event is at 11:59 on the 1st of May, and the `account.signed` event is at 12:03 on the 2nd of May.  It would start off being an incomplete flow, but get updated to its correct status when the next day's data is processed.

On the downside, it would leave more room for something going wrong between each of the steps.